### PR TITLE
Add fully-automatic mode to the Upload Widget

### PR DESCRIPTION
**Pull request by nitzanj**

### CODE REVIEW

File: sample/src/androidTest/java/cloudinary/android/sample/UploadWidgetTest.java [hunk 1]
```diff
@@ -41,16 +38,13 @@ public static void setup() throws IOException {
     public IntentsTestRule<MainActivity> intentsTestRule = new IntentsTestRule<>(MainActivity.class);
 
     // TODO: Fix UI tests for travis
-    @Ignore
+//    @Ignore
```
Comment (asisayag2) [inline sample/src/androidTest/java/cloudinary/android/sample/UploadWidgetTest.java:41]: Please remove

File: sample/src/androidTest/java/cloudinary/android/sample/UploadWidgetTest.java [hunk 2]
```diff
@@ -41,16 +38,13 @@ public static void setup() throws IOException {
...
     @Test
     public void testUploadWidget() {
-        Intent intent = new Intent();
-        intent.setData(Uri.fromFile(assetFile));
-        Instrumentation.ActivityResult result = new Instrumentation.ActivityResult(Activity.RESULT_OK, intent);
-
-        intending(hasComponent(MockNativePickerActivity.class.getName())).respondWith(result);
-        intentsTestRule.getActivity().startActivityForResult(new Intent(intentsTestRule.getActivity(), MockNativePickerActivity.class), MainActivity.CHOOSE_IMAGE_REQUEST_CODE);
-
+        UploadWidget.startActivity(intentsTestRule.getActivity(),
```
Comment (asisayag2) [inline sample/src/androidTest/java/cloudinary/android/sample/UploadWidgetTest.java:44]: Can we test the other options?

File: sample/src/androidTest/java/cloudinary/android/sample/UploadWidgetTest.java (hunk 1 above)
Comment (nitzanj) [reply sample/src/androidTest/java/cloudinary/android/sample/UploadWidgetTest.java:41]: Remove the test altogether? Or remove the `@ignore`? This test will fail every travis job until fixed.

Comment (nitzanj) [reply sample/src/androidTest/java/cloudinary/android/sample/UploadWidgetTest.java:41]: The test or the annotation? or the comment ? what state do you want this part to be in master?

Comment (asisayag2) [reply sample/src/androidTest/java/cloudinary/android/sample/UploadWidgetTest.java:41]: The comment

**Pull request merged**